### ISSUE TITLE
Allow experience key in [harm_unit] tag to be set to kill, hit, both / yes or no

### ISSUE THREAD

**Issue by Hejnewar** [wesnoth/wesnoth#8176]
### Describe the desired feature

As in the title reasons for this are:
1. With aoe attacks getting xp for each harm unit trigger can end up in granting something like 42 xp per attack (depending on aoe, level of enemies etc.)
2. Attacks that use additional [harm_unit] procs per strike end up granting way more combat experience than inteneded
3. This allows for much more flexibility when designing units without standard attacks, that rely fully on non standard way of attacking thru [harm_unit] tag

**Comment by ProditorMagnus** [comment]
https://github.com/inferno8/wesnoth-Era_of_Magic/issues/30 

> When attacker damages third party:
> 
> damager should NOT get fight xp
> damager should get kill xp
> third party should get fight xp
> When defender damages third party:
> 
> damager should NOT get fight xp
> damager should get kill xp
> third party should get fight xp

This use case is not covered by proposed values.

**Comment by Hejnewar** [comment]
>This use case is not covered by proposed values.

Yes but it covers my use cases. Nor do I have any idea how to call them, it in best case i see defender / attacker / both kill, defender / attacker / both hit, defender / attacker / both both / yes or no, which is a lot.

**Comment by ProditorMagnus** [comment]
experience= with list containing any number of `primary_kill,primary_hit,secondary_hit,secondary_kill`
With `hit` meaning `primary_hit,secondary_hit` and `kill` meaning `primary_kill,secondary_kill`
With `yes` and `no` meaning either all or none of them

**Comment by Hejnewar** [comment]
If secondary kill is not possible then there is no need for primary kill to exist, you are not excluding anything, it also doesnt seem to be needed in the EoMa case.
